### DISCUSSION TITLE
CrySPY version 1.2.4

### DISCUSSION THREAD

**Discussion by Tomoki-YAMASHITA** [Tomoki-YAMASHITA/CrySPY#60] [Announcements]
### Fixed
- bug fix
### Changed
- default value of cls_lat: equal –> random
### Added
- test version of variable composition EA (EA-vc). only binary system for now.

<hr /><em>This discussion was created from the release <a href='https://github.com/Tomoki-YAMASHITA/CrySPY/releases/tag/v1.2.4'>CrySPY version 1.2.4</a>.</em>

**Comment by yw-fang** [comment]
That's great to see this EA-vc feature!

**Reply by Tomoki-YAMASHITA** [reply]
You can try the test version of EA-vc  with this example (EMT in ASE):
https://github.com/Tomoki-YAMASHITA/CrySPY_utility/tree/master/examples/ase_Cu-Au_EA-vc

You'll get like this:
![conv_hull_gen_6](https://github.com/Tomoki-YAMASHITA/CrySPY/assets/28262881/e9631821-9d9a-457e-806d-7dc8bd58c646)

**Comment by yw-fang** [comment]
That's a pretty nice feature. Did you get the convex hull directly from CrySPY or obtained by post-processing the energy data?

**Reply by Tomoki-YAMASHITA** [reply]
CrySPY automatically draws the convex hull every generation.
You don't have to do it.

**Comment by yw-fang** [comment]
By the way, have you maintained a list anywhere we can append the publications using CrySPY?
In one of our recent works, we have used it: https://www.nature.com/articles/s43246-024-00500-9 
In our group, I am very sure more than 3 members are using CrySPY now, and it's expected we will use it further. Actually one of the members asked me about the availability of varied compositions search in CrySPY only two weeks ago.

**Reply by Tomoki-YAMASHITA** [reply]
I haven't maintained, but I check it because I receive notifications (google scholar) when our paper is cited.
I have checked yours. Congratulations on your publication.
Creating a page of related work on the documentation site might be a good idea.

I look forward to next publication from your group.

Lately, we have developed EA-vc for ternary systems or higher.

**Reply by yw-fang** [reply]
Get it! Thanks for your words! For the multinary systems, is it available now in any of the branches?

**Reply by Tomoki-YAMASHITA** [reply]
EA-vc for binary systems is already implemented in CrySPY 1.2.4, but I haven't tested anything but  calc_code = ASE.
I guess it would work fine with VASP and QE.

**Reply by yw-fang** [reply]
My former question might have caused some confusion to you. What I inquired about was the the implementation beyond ternary, is it available now?

**Reply by Tomoki-YAMASHITA** [reply]
Not yet implemented!
Currently we are working on calculating the convex hull and hull distance in multi dimensional systems.
